src/cmd/services/m3aggregator/config/aggregator.go

```diff
@@ -152,6 +156,78 @@ type AggregatorConfiguration struct {
 	EntryPool pool.ObjectPoolConfiguration `yaml:"entryPool"`
 }
 
+// InstanceIDType is the instance ID type that defines how the
+// instance ID is constructed, which is then used to lookup the
+// aggregator instance in the placement.
+type InstanceIDType uint
+
+const (
+	// HostIDPortInstanceIDType specifies to use the host ID
+	// concatenated with the port to be used for lookup
+	// in the placement.
+	// NB: this is a legacy instance ID type and is how the instance
+	// ID used to be constructed which imposed the strange
+	// requirement that the instance ID in the topology used to require
+	// the port concat'd with the host ID).
+	HostIDPortInstanceIDType InstanceIDType = iota
+	// HostIDInstanceIDType specifies to just use the host ID
+	// as the instance ID for lookup in the placement.
+	HostIDInstanceIDType
+
+	// defaultInstanceIDType must be used as the legacy instance ID
+	// since the config needs to be backwards compatible and for those
+	// not explicitly specifying the instance ID type it will cause
+	// existing placements to not work with latest versions of the aggregator
+	// in a backwards compatible fashion.
+	defaultInstanceIDType = HostIDPortInstanceIDType
+)
+
+func (t InstanceIDType) String() string {
+	switch t {
+	case HostIDInstanceIDType:
+		return "host_id"
+	case HostIDPortInstanceIDType:
+		return "host_id_port"
+	}
+	return "unknown"
+}
+
+var (
+	validInstanceIDTypes = []InstanceIDType{
+		HostIDInstanceIDType,
+		HostIDPortInstanceIDType,
+	}
+)
+
+// UnmarshalYAML unmarshals a InstanceIDType into a valid type from string.
+func (t *InstanceIDType) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var str string
+	if err := unmarshal(&str); err != nil {
+		return err
+	}
+	if str == "" {
+		*t = defaultInstanceIDType
+		return nil
+	}
+	strs := make([]string, 0, len(validInstanceIDTypes))
```

Comment:
Maybe better to make validInstanceIDTypes a `[]string{HostIDInstanceIDType.String(), HostIDPortInstanceIDType.String()}` to avoid allocing a new array here?

Reply:
Since it's during config time I think this isn't necessary, I don't necessarily want to change how this is done everywhere else right at this time if that's ok, heh.